src/parquet/column/reader.h

```diff
@@ -155,50 +155,59 @@ inline size_t TypedColumnReader<TYPE>::ReadValues(size_t batch_size, T* out) {
 }
 
 template <int TYPE>
-inline size_t TypedColumnReader<TYPE>::ReadBatch(int batch_size, int16_t* def_levels,
+inline size_t TypedColumnReader<TYPE>::ReadBatch(int32_t batch_size, int16_t* def_levels,
     int16_t* rep_levels, T* values, size_t* values_read) {
-  // HasNext invokes ReadNewPage
-  if (!HasNext()) {
-    *values_read = 0;
-    return 0;
-  }
-
-  // TODO(wesm): keep reading data pages until batch_size is reached, or the
-  // row group is finished
-  batch_size = std::min(batch_size, num_buffered_values_);
-
-  size_t num_def_levels = 0;
-  size_t num_rep_levels = 0;
-
-  size_t values_to_read = 0;
-
-  // If the field is required and non-repeated, there are no definition levels
-  if (descr_->max_definition_level() > 0) {
-    num_def_levels = ReadDefinitionLevels(batch_size, def_levels);
-    // TODO(wesm): this tallying of values-to-decode can be performed with better
-    // cache-efficiency if fused with the level decoding.
-    for (size_t i = 0; i < num_def_levels; ++i) {
-      if (def_levels[i] == descr_->max_definition_level()) {
-        ++values_to_read;
+  *values_read = 0;
+  size_t total_values = 0;
+  int32_t level_offset = 0;
+  int32_t value_offset = 0;
+  size_t current_values_read = 0;
```

Comment:
The parquet types use `int32_t`. I can use `int` too. I don't have a strong opinion.